hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieLockConfig.java

```diff
@@ -108,6 +109,12 @@ public class HoodieLockConfig extends HoodieConfig {
       .sinceVersion("0.8.0")
```

Comment:
I have an idea, wdyt about setting the `metatable`'s subpath as the default path?

Reply:
> I have an idea, wdyt about setting the `metatable`'s subpath as the default path?

yeah, I've thought about it too, if so, since we cannot get basePath from lock config, need to pass extra parameters into the constructor, and other lock providers also need to be refactored. What do you think, is it worth ?

Reply:
Yes, we need to get the table's base path so that we can call `HoodieTableMetadata#getMetadataTableBasePath`. 

Let me think about it.

Reply:
I see the `LockConfiguration` takes all the config items in `HoodieWriteConfig`:

`this.lockConfiguration = new LockConfiguration(writeConfig.getProps());`

in `LockManager`

Maybe we can extract the base path from it？